Flaky unit test: TestReconcileForCluster

**Which jobs are flaking:**

On some CI runs for PRs, the `TestReconcileForCluster` spec is failing, but often succeeds on `/retest`.

**Which tests are flaking:**

`TestReconcileForCluster/do_not_reconcile_for_a_paused_cluster`

```shell
ok  	sigs.k8s.io/cluster-api-addon-provider-helm/controllers	9.075s
--- FAIL: TestReconcileForCluster (0.00s)
    --- FAIL: TestReconcileForCluster/do_not_reconcile_for_a_paused_cluster (0.00s)
        helmchartproxy_controller_phases_test.go:321: 
            Expected
                <bool>: true
            to be false
FAIL
FAIL	sigs.k8s.io/cluster-api-addon-provider-helm/controllers/helmchartproxy	0.114s
ok  	sigs.k8s.io/cluster-api-addon-provider-helm/controllers/helmreleaseproxy	0.055s
FAIL
make: *** [Makefile:395: test] Error 1
```

**Testgrid link:**

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-addon-provider-helm/203/pull-cluster-api-addon-provider-helm-test-main/1782485852740390912

**Reason for failure (if possible):**

?

**Anything else we need to know:**
- This test was added in #189 and has been flaky since it merged AFAICT.


/kind flake

--- a/controllers/helmchartproxy/helmchartproxy_controller_phases_test.go
+++ b/controllers/helmchartproxy/helmchartproxy_controller_phases_test.go
@@ -313,7 +313,7 @@ func TestReconcileForCluster(t *testing.T) {
 		},
 		{
 			name:                          "do not reconcile for a paused cluster",
-			helmChartProxy:                fakeReinstallHelmChartProxy,
+			helmChartProxy:                fakeHelmChartProxy1,
 			existingHelmReleaseProxy:      fakeHelmReleaseProxy,
 			cluster:                       fakeClusterPaused,
 			expectHelmReleaseProxyToExist: false,

